src/Microsoft.Extensions.Caching.SqlConfig.Tools/Program.cs

```diff
@@ -15,23 +15,32 @@ public class Program
         private string _connectionString = null;
         private string _schemaName = null;
         private string _tableName = null;
+        private readonly IConsole _console;
+        private IReporter _reporter;
 
-        private readonly ILogger _logger;
-
-        public Program()
+        public Program(IConsole console)
         {
-            var loggerFactory = new LoggerFactory();
-            loggerFactory.AddConsole();
-            _logger = loggerFactory.CreateLogger<Program>();
+            Ensure.NotNull(console, nameof(console));
+
+            _console = console;
         }
 
         public static int Main(string[] args)
         {
-            return new Program().Run(args);
+            return new Program(PhysicalConsole.Singleton).Run(args);
         }
 
         public int Run(string[] args)
         {
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            _reporter = new ReporterBuilder()
+                .WithConsole(_console)
+                .Verbose(f => f.WithColor(ConsoleColor.DarkGray))
+                .Warn(f => f.WithColor(ConsoleColor.Yellow))
+                .Error(f => f.WithColor(ConsoleColor.Red))
+                .Build();
```

Comment:
This tool doesn't actually have `--verbose` flags. Though I suppose detecting output redirection for colorless output is still valid...although probably a very unlikely usage.